src/hardware/openbench-logic-sniffer/protocol.c

```diff
@@ -527,3 +527,155 @@ SR_PRIV int ols_receive_data(int fd, int revents, void *cb_data)
+			sr_dbg("Setting OLS stage %d trigger.", i);
+			if ((ret = set_basic_trigger_stage(&basic_trigger_desc, serial, i)) != SR_OK)
+				return ret;
+		}
+	} else {
+		/* No triggers configured, force trigger on first stage. */
+		sr_dbg("Forcing trigger at stage 0.");
+		if ((ret = set_basic_trigger_stage(&basic_trigger_desc, serial, 0)) != SR_OK)
+			return ret;
+		delaycount = readcount;
+	}
+
+	/* Samplerate. */
+	sr_dbg("Setting samplerate to %" PRIu64 "Hz (divider %u)",
+			devc->cur_samplerate, devc->cur_samplerate_divider);
+	arg[0] = devc->cur_samplerate_divider & 0xff;
+	arg[1] = (devc->cur_samplerate_divider & 0xff00) >> 8;
+	arg[2] = (devc->cur_samplerate_divider & 0xff0000) >> 16;
+	arg[3] = 0x00;
+	if (send_longcommand(serial, CMD_SET_DIVIDER, arg) != SR_OK)
+		return SR_ERR;
+
+	/* Send sample limit and pre/post-trigger capture ratio. */
+	sr_dbg("Setting sample limit %d, trigger point at %d",
+			(readcount - 1) * 4, (delaycount - 1) * 4);
+
+	if (devc->max_samples > 256 * 1024) {
+		arg[0] = ((readcount - 1) & 0xff);
+		arg[1] = ((readcount - 1) & 0xff00) >> 8;
+		arg[2] = ((readcount - 1) & 0xff0000) >> 16;
+		arg[3] = ((readcount - 1) & 0xff000000) >> 24;
+		if (send_longcommand(serial, CMD_CAPTURE_READCOUNT, arg) != SR_OK)
+			return SR_ERR;
+		arg[0] = ((delaycount - 1) & 0xff);
+		arg[1] = ((delaycount - 1) & 0xff00) >> 8;
+		arg[2] = ((delaycount - 1) & 0xff0000) >> 16;
+		arg[3] = ((delaycount - 1) & 0xff000000) >> 24;
+		if (send_longcommand(serial, CMD_CAPTURE_DELAYCOUNT, arg) != SR_OK)
+			return SR_ERR;
+	} else {
+		arg[0] = ((readcount - 1) & 0xff);
+		arg[1] = ((readcount - 1) & 0xff00) >> 8;
+		arg[2] = ((delaycount - 1) & 0xff);
+		arg[3] = ((delaycount - 1) & 0xff00) >> 8;
+		if (send_longcommand(serial, CMD_CAPTURE_SIZE, arg) != SR_OK)
+			return SR_ERR;
+	}
+
+	/* Flag register. */
+	sr_dbg("Setting intpat %s, extpat %s, RLE %s, noise_filter %s, demux %s, %s clock%s",
+			devc->capture_flags & CAPTURE_FLAG_INTERNAL_TEST_MODE ? "on": "off",
+			devc->capture_flags & CAPTURE_FLAG_EXTERNAL_TEST_MODE ? "on": "off",
+			devc->capture_flags & CAPTURE_FLAG_RLE ? "on" : "off",
+			devc->capture_flags & CAPTURE_FLAG_NOISE_FILTER ? "on": "off",
+			devc->capture_flags & CAPTURE_FLAG_DEMUX ? "on" : "off",
+			devc->capture_flags & CAPTURE_FLAG_CLOCK_EXTERNAL ? "external" : "internal",
+			devc->capture_flags & CAPTURE_FLAG_CLOCK_EXTERNAL ? (devc->capture_flags & CAPTURE_FLAG_INVERT_EXT_CLOCK
+				? " on falling edge" : "on rising edge") : "");
+
+	/*
+	 * Enable/disable OLS channel groups in the flag register according
+	 * to the channel mask. 1 means "disable channel".
+	 */
+	devc->capture_flags &= ~0x3c;
+	devc->capture_flags |= ~(changroup_mask << 2) & 0x3c;
+
+	/* RLE mode is always zero, for now. */
+
+	arg[0] = devc->capture_flags & 0xff;
+	arg[1] = devc->capture_flags >> 8;
+	arg[2] = arg[3] = 0x00;
+	if (send_longcommand(serial, CMD_SET_FLAGS, arg) != SR_OK)
+		return SR_ERR;
+
+	return SR_OK;
+}
```

Comment:
I assume these two functions were just moved and not modified, right?
In general, these changes look fine to me. Maybe again a short sentence in the commit message that this is in preparation for more complex triggers, but no further comments.

Reply:
Yes. I added a comment about this.